packages/devtools_app/lib/src/utils.dart

```diff
@@ -1069,3 +1069,54 @@ String prettyTimestamp(
   );
   return DateFormat.Hms().format(timestampDT); // HH:mm:ss
 }
+
+/// A [ChangeNotifier] that holds a list of data.
+///
+/// When [value] is replaced with something that is not equal to the old
+/// value as evaluated by the equality operator ==, this class notifies its
+/// listeners.
+///
+/// This class also exposes methods `addAndNotify` and `addAllAndNotify` that
+/// will notify listeners upon the list [value] being modified.
+class ListValueNotifier<T extends List<V>, V> extends ChangeNotifier
+    implements ValueListenable<T> {
+  /// Creates a [ChangeNotifier] that wraps this value.
+  ListValueNotifier(this._value);
+
+  /// The current value stored in this notifier.
+  ///
+  /// When the value is replaced with something that is not equal to the old
+  /// value as evaluated by the equality operator ==, this class notifies its
+  /// listeners.
+  @override
+  T get value => _value;
+  T _value;
+  set value(T newValue) {
```

Comment:
probably avoid providing a set value method here as it will be tricky to make work efficiently with the other apis. You might instead provide a clear() method.

Reply:
or add 
```dart
void initialize(List<V>  elements) {
}
```
to make it clear the the actual value returned will be a view of the list rather than an identical object.

Reply:
this was removed before your review - may want to peek at the updated bits. I changed the class to extend ValueNotifier instead of ChangeNotifier, so this copied override was removed.